Assets/Scripts/Entity/Enemy/EnemyDrop.cs

```diff
@@ -26,12 +35,34 @@ void SubscribeToEnemyDeath()
         }
     }
 
+    void PopulateDropList()
+    {
+        for(int i = 0; i < heartPrefabs.Count; i++)
+        {
+            var dropObject = new Drop
```

Comment:
var master race. JS will rule us all. I am of the opinion that in C# you only ever need to explicitly declare type if it could possibly be ambiguous. In this case you can see on the same line exactly what the type will be. Additionally var isn't like var in JS, it's not an actual type and it is converted into the corresponding type at compile time, not runtime which means there is no performance hit.

Reply:
Biggest (and I think only) issue with `var` is readability. I absolutely detest JS because when you get large chunks of code, unless the variable names are really verbose, you have no idea what each variable is. In fact, you pretty accept the fact that you don't know what a variable is and just "assume" it's of a type you can work with. This is what makes JS so bad. People come in and just start doing what ever they want, adding properties to objects at runtime, adding methods to objects at runtime, that by the time you get the object, only god knows what is in it and where it has been.

JS aside, `var` is still bad because of the reason for large chunks of code. Sure here, it can be obvious that `dropObject = new Drop` is a Drop object. But there simply is no need, you literally save typing 1 letter, and you gain a good deal of readability. Specially if you work with more complex design patterns such as factories, or you use polymorphism. 

General rule of thumb, you avoid `var` if possible. You never gain anything, and you will usually lose quite a bit of readability.